enumerables.rb

```diff
@@ -0,0 +1,114 @@
+module Enumerable
+  def my_each()
+    i = 0
+    while i < size
+      return to_enum unless block_given?
+
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      return to_enum unless block_given?
+
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    selected_array = []
+    i = 0
+    while i < size
+      return to_enum unless block_given?
+
+      selected_array << self[i] if yield(self[i])
+      i += 1
+    end
+    selected_array
+  end
+
+  def my_all?
+    i = 0
+    while i < size
+      return false if block_given? && !yield(self[i])
+      return false if self[i] == false || self[i].nil?
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?
+    i = 0
+    while i < size
+      return true if block_given? && yield(self[i])
+      return true if !self[i] == false && !self[i].nil?
+
+      i += 1
+    end
+    false
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil
```ruby
true_array = [nil, false, true, []] 
false_array = [nil, false, nil, false]
true_array.my_any? == true_array.any? #true
```
- [ ]  when a class is passed as an argument returns true if at least one of the collection is a member of such class
```ruby
words = %w[dog door rod blade]
words.my_any?(Integer) == words.any?(Integer) #true
```
- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex
```ruby
words.my_any?(/z/) == words.any?(/z/) #true
```
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```ruby
words.my_any?('cat') == words.any?('cat') #true